corehq/blobs/mixin.py

```diff
@@ -261,34 +315,41 @@ class BlobHelper(object):
     not recommended while it is wrapped in this class.
     """
 
-    def __init__(self, doc, database):
+    def __init__(self, doc, database, type_code, domain=None):
         if doc.get("_id") is None:
             raise TypeError("BlobHelper requires a real _id")
         self._id = doc["_id"]
         self.doc = doc
         self.doc_type = doc["doc_type"]
+        if "domain" in doc:
+            self.domain = doc["domain"]
+        elif self.doc_type == "Domain":
+            self.domain = doc["name"]
+        self._blobdb_type_code = type_code
         self.database = database
         self.couch_only = "external_blobs" not in doc
         self._migrating_blobs_from_couch = bool(doc.get("_attachments")) \
             and not self.couch_only
         self._attachments = doc.get("_attachments")
-        blobs = self.get_external_blobs(doc, {})
-        self.external_blobs = {n: BlobMeta.wrap(m.copy())
+        blobs = self.get_external_blobs(doc, self.database.dbname, {})
+        self.external_blobs = {n: BlobMetaPointer.wrap(m.copy())
                                for n, m in six.iteritems(blobs)}
 
     _atomic_blobs = None
 
     @staticmethod
-    def get_external_blobs(doc, default=None):
-        return doc.get("external_blobs", default)
+    def get_external_blobs(doc, dbname, default=None):
+        blobs = doc.get("external_blobs", default)
+        if blobs is default:
+            return blobs
+        doc_id = doc["_id"]
+        return {n: BlobMetaPointer._normalize_json(dbname, doc_id, m)
+                for n, m in six.iteritems(blobs)}
```

Comment:
Can this be
```
if "external_blobs" not in doc:
    return default
doc_id = doc["_id"]
return {n: BlobMetaPointer._normalize_json(dbname, doc_id, m) for n, m in six.iteritems(blobs)}
```

Reply:
That wouldn't work because `six.iteritems(blobs)` would cause a NameError

Reply:
I agree with @mkangia that his version requires less brain power to understand. Also I think you're missing the point, you'd do `six.iteritems(doc['external_blobs'])`

Reply:
This method (`get_external_blobs`) got refactored and doesn't exist anymore.

Reply:
thats even better :)